corehq/apps/settings/tests/test_views.py

```diff
@@ -105,29 +107,39 @@ def _create_view_for_user(self, user):
 
 
 @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=True)
-class TwoFactorSetupView_FormKwargs_Tests(SimpleTestCase):
+class TwoFactorSetupView_RegisteredMethods_Tests(SimpleTestCase):
+
     @override_settings(ALLOW_PHONE_AS_DEFAULT_TWO_FACTOR_DEVICE=False)
-    def test_phone_methods_are_prohibited_when_settings_are_disabled(self):
+    def test_phone_methods_are_not_registered_when_settings_are_disabled(self):
         user = self._create_user(belongs_to_messaging_domain=True)
-        view = self._create_view_for_user(user)
-        self.assertFalse(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator'], [m.code for m in registry.get_methods()])
 
-    def test_phone_methods_are_allowed_when_user_belongs_to_messaging_domain(self):
+    def test_phone_methods_are_registered_when_user_belongs_to_messaging_domain(self):
         user = self._create_user(belongs_to_messaging_domain=True)
-        view = self._create_view_for_user(user)
-        self.assertTrue(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator', 'sms', 'call'], [m.code for m in registry.get_methods()])
 
-    def test_phone_methods_are_prohibited_when_user_does_not_belongs_to_messaging_domain(self):
+    def test_phone_methods_are_not_registered_when_user_does_not_belongs_to_messaging_domain(self):
         user = self._create_user(belongs_to_messaging_domain=False)
-        view = self._create_view_for_user(user)
-        self.assertFalse(view.get_form_kwargs(step='method')['allow_phone_2fa'])
+        self._create_view_for_user(user)
+        self.assertCountEqual(['generator'], [m.code for m in registry.get_methods()])
+
+    def test_custom_methods_are_registsered(self):
+        user = self._create_user(belongs_to_messaging_domain=True)
+        self._create_view_for_user(user)
+
+        generator_method = registry.get_method('generator')
+        self.assertEqual(HQGeneratorMethod, type(generator_method))
+
+        call_method = registry.get_method('call')
+        self.assertEqual(HQPhoneCallMethod, type(call_method))
+
+        sms_method = registry.get_method('sms')
+        self.assertEqual(HQSMSMethod, type(sms_method))
```

Comment:
Nit: I think our assertions are generally of the `self.assert(actual, expected)` form. Flipping them to `self.assertEqual(type(generator_method), HQGeneratorMethod)` might make them slightly easier to understand that way